rest/data/commit_queue.go

```diff
@@ -602,7 +602,23 @@ func CheckCanRemoveCommitQueueItem(ctx context.Context, sc Connector, usr *user.
 				Message:    errors.Wrapf(err, "finding patch '%s'", itemId).Error(),
 			}
 		}
-		if usr.Id != utility.FromStringPtr(patch.Author) {
+
+		// TODO: Remove user lookup and OnlyAPI condition after EVG-20118 is completed.
+		// Since we don't require users to save their GitHub information before submitting patches,
+		// we must allow them to remove any patches created by service users (OnlyAPI = true).
+		patchAuthor := utility.FromStringPtr(patch.Author)
+		patchUsr, err := user.FindOneById(patchAuthor)
+		if patchUsr == nil || err != nil {
```

Comment:
We handle these cases in separate if statements because errors.Wrapf defaults to nil if error isn't nil, and so nil.Error will panic (in this case I'd also expect different error messages -- one is an error finding, one is that user isn't found).

Reply:
oops, thanks annie!